src/algebra/field_power.lean

```diff
@@ -148,28 +158,106 @@ begin
+lemma fpow_le_one_of_nonpos {p : K} (hp : 1 ≤ p) {z : ℤ} (hz : z ≤ 0) : p ^ z ≤ 1 :=
 calc p ^ z ≤ p ^ 0 : fpow_le_of_le hp hz
           ... = 1        : by simp
 
-lemma one_le_fpow_of_nonneg {p : α} (hp : 1 ≤ p) {z : ℤ} (hz : 0 ≤ z) : 1 ≤ p ^ z :=
+lemma one_le_fpow_of_nonneg {p : K} (hp : 1 ≤ p) {z : ℤ} (hz : 0 ≤ z) : 1 ≤ p ^ z :=
 calc p ^ z ≥ p ^ 0 : fpow_le_of_le hp hz
           ... = 1        : by simp
 
 end ordered_field_power
 
-lemma one_lt_pow {α} [linear_ordered_semiring α] {p : α} (hp : 1 < p) : ∀ {n : ℕ}, 1 ≤ n → 1 < p ^ n
+lemma one_lt_pow {K} [linear_ordered_semiring K] {p : K} (hp : 1 < p) : ∀ {n : ℕ}, 1 ≤ n → 1 < p ^ n
 | 1 h := by simp; assumption
 | (k+2) h :=
   begin
-    rw ←one_mul (1 : α),
+    rw ←one_mul (1 : K),
     apply mul_lt_mul,
     { assumption },
     { apply le_of_lt, simpa using one_lt_pow (nat.le_add_left 1 k)},
     { apply zero_lt_one },
     { apply le_of_lt (lt_trans zero_lt_one hp) }
   end
 
-lemma one_lt_fpow {α}  [discrete_linear_ordered_field α] {p : α} (hp : 1 < p) :
+lemma one_lt_fpow {K}  [discrete_linear_ordered_field K] {p : K} (hp : 1 < p) :
   ∀ z : ℤ, 0 < z → 1 < p ^ z
 | (int.of_nat n) h := one_lt_pow hp (nat.succ_le_of_lt (int.lt_of_coe_nat_lt_coe_nat h))
+
+section ordered
+variables  {K : Type*} [discrete_linear_ordered_field K]
+
+lemma nat.fpow_pos_of_pos {p : ℕ} (h : 0 < p) (n:ℤ) : (p:K)^n > 0 :=
+by { apply fpow_pos_of_pos, exact_mod_cast h }
+
+lemma nat.fpow_ne_zero_of_pos {p : ℕ} (h : 0 < p) (n:ℤ) : (p:K)^n ≠ 0 :=
+ne_of_gt (nat.fpow_pos_of_pos h n)
+
+lemma fpow_strict_mono {x : K} (hx : 1 < x) :
+  strict_mono (λ n:ℤ, x ^ n) :=
+λ m n h, show x ^ m < x ^ n,
+begin
+  have xpos : 0 < x := by linarith,
+  have h₀ : x ≠ 0 := by linarith,
+  have hxm : 0 < x^m := fpow_pos_of_pos xpos m,
+  have hxm₀ : x^m ≠ 0 := ne_of_gt hxm,
+  suffices : 1 < x^(n-m),
+  { replace := mul_lt_mul_of_pos_right this hxm,
+    simpa [*, fpow_add, mul_assoc, fpow_neg, inv_mul_cancel], },
+  apply one_lt_fpow hx, linarith,
+end
+
+@[simp] lemma fpow_mono {x : K} (hx : 1 < x) {m n : ℤ} :
+  x ^ m < x ^ n ↔ m < n :=
+(fpow_strict_mono hx).lt_iff_lt
+
+@[simp] lemma fpow_inj {x : K} (h₀ : 0 < x) (h₁ : x ≠ 1) {m n : ℤ} :
+  x ^ m = x ^ n ↔ m = n :=
+begin
+  split; intro h, swap, {simp [h]},
+  rcases lt_trichotomy x 1 with H|rfl|H,
+  { apply (fpow_strict_mono (one_lt_inv h₀ H)).injective,
+    show x⁻¹ ^ m = x⁻¹ ^ n,
+    rw [← fpow_inv, ← fpow_mul, ← fpow_mul, mul_comm _ m, mul_comm _ n, fpow_mul, fpow_mul, h], },
+  { contradiction },
+  { exact (fpow_strict_mono H).injective h, },
+end
+
+end ordered
+
+section
+variables {K : Type*} [discrete_field K]
+
+@[simp] theorem fpow_neg_mul_fpow_self (n : ℕ) {x : K} (h : x ≠ 0) :
+  x^-(n:ℤ) * x^n = 1 :=
+begin
+  convert inv_mul_cancel (pow_ne_zero n h),
+  rw [fpow_neg, one_div_eq_inv, fpow_of_nat]
+end
+
+@[simp, move_cast] theorem cast_fpow [char_zero K] (q : ℚ) (k : ℤ) :
```

Comment:
Can this be proven with the `discrete_field` version of `map_fpow`

Reply:
Good idea! Done.